chainerx_cc/chainerx/native/native_device/dot.cc

```diff
@@ -166,16 +177,31 @@ void NativeDevice::Dot(const Array& a, const Array& b, const Array& out) {
         CHAINERX_ASSERT(out.shape()[0] == m);
         CHAINERX_ASSERT(out.shape()[1] == n);
 
+        using AccT = typename std::conditional<std::is_same<T, Float16>{}, float, T>::type;
```

Comment:
Another nitpick: we can use `std::conditonal_t` because we already require C++14.